src/hardware/sblaster.cpp

```diff
@@ -819,7 +819,12 @@ static void dsp_dma_callback(const DmaChannel* chan, const DmaEvent event)
 			// ready for playback. This is when we set the callback running to
 			// play the data.
 			sblaster->MaybeWakeup();
-			callback_timing.SetPerTick();
+
+			constexpr auto PerTickCountThreshold = 4;
```

Comment:
Any reason for picking the number 4? Could you explain in a comment?

Reply:
Yup; added a comment and used a bit better name:

```c++
// If the DMA transfer is setup with a base count of fewer than
// three elements (which is four bytes given one 16-bit stereo frame
// held in an 8-bit DMA channel), then we know the software intends
// to overwrite the DMA content on the fly instead of pre-generating
// large chunks of DMA audio. In these cases we prefer the
// fine-grained per-frame callback. (The minus one is because DMA
// counts are in addition to one; so a base count of zero is one
// element).

constexpr auto MaxSingleFrameBaseCount = sizeof(int16_t) * 2 - 1;

(chan->base_count <= MaxSingleFrameBaseCount)
        ? callback_type.SetPerFrame()
        : callback_type.SetPerTick();
```

Reply:
Awesome!